src/components/cart/Amount.jsx

```diff
@@ -0,0 +1,54 @@
+import { useEffect, useState } from "react";
+import { useProducts } from "../../context/product-context";
+
+function Amount() {
+  const { state } = useProducts();
+  const { cart } = state;
+  const [amount, setAmount] = useState({ totalMRP: 0, totalDiscount: 0 });
+  const {totalMRP, totalDiscount} = amount
+  useEffect(() => {
+    setAmount(
+      cart.reduce(
+        (acc, {price, qty}) => ({
+          ...acc,
+          totalMRP: acc.totalMRP + price.original * qty,
+          totalDiscount:
+            acc.totalDiscount + price.discounted * qty,
+        }),
+        { totalMRP: 0, totalDiscount: 0 }
+      )
+    );
+  }, [cart]);
```

Comment:
Suggestion: This doesn't look like a side-effect so you can do this in a simple function.

Reply:
I wanted to change my amount every time my cart item changes. So I put it in useEffect and gave the dependency of cart.